IngresseSDK/Parser/URLBuilder.swift

```diff
@@ -80,7 +80,8 @@ public class URLBuilder: NSObject {
         return self
     }
     
-    public func addParameter(key: String, value: Any) -> URLBuilder {
+    public func addParameter(key: String, value: Any?) -> URLBuilder {
+        guard let value = value else { return self }
         self.parameters[key] = "\(value)"
         
         return self
```

Comment:
Esse era o método que eu estava falando e pra isso usar o **Encodable**
```suggestion
    }

    func addEncodableParameter( _ param: Encodable) -> URLBuilder {

        var builder = self
        param.encoded?.forEach { builder = builder.addParameter(key: $0, value: $1) }
        return builder
    }
```

Eu criei uma extension do **Encodable** também para ele converter para um **Dictionary<String, nil>**

```swift
extension Encodable {

    var encoded: [String: Any]? {

        do {

            let data = try JSONEncoder().encode(self)
            let dictionaryAny = try JSONSerialization.jsonObject(with: data,
                                                                 options: .allowFragments)
            return dictionaryAny as? [String: Any]
        } catch {

            return nil
        }
    }
}
```

Assim, você não precisa ficar colocando as `keys` em vários lugares